### DISCUSSION TITLE
Can't connect to my Homespan

### DISCUSSION THREAD

**Discussion by Fergtato** [HomeSpan/HomeSpan#273] [Q&A] [answered]
I've set this all up on my ESP32 following the README but when I try to connect to the device in the Homekit app using the default code I keep getting 

> "Unable to Add Accessory - Failed with OSStatus code -6700 (kUnknownErr)"

My log outputs also look like this:

> 23:12:37.065 -> *** ERROR:  Bad GET request - URL not found
> 23:12:37.065 -> 
> 23:12:42.075 -> 
> 23:12:42.075 -> *** ERROR:  Bad GET request - URL not found
> 23:12:42.075 -> 
> 23:12:47.068 -> 
> 23:12:47.068 -> *** ERROR:  Bad GET request - URL not found
> 23:12:47.129 -> 
> 23:12:52.015 -> 
> 23:12:52.015 -> *** ERROR:  Bad GET request - URL not found
> 23:12:52.015 -> 
> 23:12:58.036 -> 
> 23:12:58.036 -> *** ERROR:  Bad GET request - URL not found

Every few seconds it's just printing this even though it connected to my network successfully...

> 23:07:36.033 -> Accessory ID:      25:17:9A:3E:08:43                               LTPK: 7CB2B9C66CAE22CE7F2FA6FEB0E23609180D65A07C3D669D56842C5961A66717
> 23:07:36.033 -> No Paired Controllers
> 23:07:36.033 -> 
> 23:07:36.033 -> Accessory configuration number: 1
> 23:07:36.033 -> 
> 23:07:36.033 -> HomeSpan LEDs is READY!
> 23:07:36.033 -> 
> 23:07:36.033 -> Trying to connect to Skynet.  Waiting 1 second(s) for response...
> 23:07:37.138 -> Trying to connect to Skynet.  Waiting 2 second(s) for response...
> 23:07:39.134 -> Trying to connect to Skynet.  Waiting 4 second(s) for response...
> 23:07:41.749 -> Successfully connected to Skynet! IP Address: 192.168.0.95
> 23:07:41.749 -> 
> 23:07:41.749 -> Starting MDNS...
> 23:07:41.749 -> 
> 23:07:41.749 -> HostName:      HomeSpan-25179A3E0843.local:80
> 23:07:41.749 -> Display Name:  HomeSpan LEDs
> 23:07:41.749 -> Model Name:    HomeSpan-ESP32
> 23:07:41.749 -> Setup ID:      HSPN
> 23:07:41.749 -> 
> 23:07:41.749 -> Starting HAP Server on port 80 supporting 8 simultaneous HomeKit Controller Connections...
> 23:07:41.749 -> 
> 23:07:41.749 -> DEVICE NOT YET PAIRED -- PLEASE PAIR WITH HOMEKIT APP
> 23:07:41.749 -> 
> 23:07:42.254 -> 
> 23:07:42.254 -> *** ERROR:  Bad GET request - URL not found
> 23:07:42.254 -> 
> 23:07:47.048 -> 
> 23:07:47.048 -> *** ERROR:  Bad GET request - URL not found

Any help would be appreciated!

**Comment by Xorlent** [comment]
Did you set up a HomeKit hub?

https://support.apple.com/en-us/HT207057

**Reply by Fergtato** [reply]
Yes I have my iPad as a homekit hub

**Reply by Xorlent** [reply]
Hmm.  Is it possible the HomeKit hub and the device you're trying to pair are on separate WiFi SSIDs, or maybe you have intra-SSID traffic disabled on the router/firewall you are using?  Broadcast traffic also needs to be permitted, as that is how mDNS works.

**Reply by Fergtato** [reply]
No they're deffinitely on the same network, and I have a few other devices using homekit on the same network. Is there something I need to look into maybe in my routers settings then?

**Reply by Xorlent** [reply]
If you have not already, I'd recommend switching to log level 2 (in serial monitor, just type "L2") and see what is going on with the traffic.  Hopefully it helps give you at least some clue as to what is going wrong.

**Reply by Fergtato** [reply]
Typing L2 seems to just freeze it... logs stop outputting and no commands work anymore... hmmm

**Reply by Xorlent** [reply]
Well in that case, let's see your INO file!

**Comment by HomeSpan** [comment]
Even if on the same SSID, are you using a router that has both 2.4 and 5 GHz bands?  If so, note that the ESP32 only works on the 2.4 GHz band, which means your router needs to seamlessly transmit across the two.  Some routers are configured to keep traffic separate, which will only allow HomeSpan to work if your iPhone and other HomeKit devices are also restricted to using just the 2.4 GHz band.

From some research I did on the web, it does not look like this is a Homespan issue, but something more general to HomeKit that others have seen under different circumstances.

**Reply by Fergtato** [reply]
Yes I am using a router that has 5ghz and 2.4ghz so this could well be the problem. I'm looking through my routers settings now to see if this is something I can change but can't seem to find anything. Will update if I figure it out.
Thanks!

**Comment by HomeSpan** [comment]
A problem with the connection should not impact the Serial Monitor.  Are you starting with Example 1 in the tutorial - just a simple LightBulb with no pins, no LEDs, no switches, etc.

**Reply by Fergtato** [reply]
I'm starting with the dimmable LED tutorial, every other command works fine until I type L2 and it just stops giving feedback

**Reply by Fergtato** [reply]
Actually now every command is breaking it... what on earth is going on

**Comment by HomeSpan** [comment]
Start with Example 1.  Once booted, issue an 'H' command from the CLI.  This will erase the current Device ID and restart in an unpaired mode.  Then try re-pairing with the Home App.

**Reply by Fergtato** [reply]
Even typing H with example 1 seems to do nothing now... I had this working before! I might try a different board

**Comment by HomeSpan** [comment, marked as answer]
It's always possible it's the board.  Worth trying another if you have.  What board are you using?

**Reply by Fergtato** [reply]
ESP32 Dev Board I believe, trying a fresh one now

**Reply by Fergtato** [reply]
YEP! New board fixed the problem, ugh! Thanks for all the help anyway

**Reply by HomeSpan** [reply]
That's great.  It's rarely a problem with a board, or a cable.  But sometimes it is!  Glad it's now working.
